core/src/cats/sql_create.cc

```diff
@@ -102,21 +102,25 @@ bool BareosDb::CreateJobRecord(JobControlRecord* jcr, JobDbRecord* jr)
  */
 bool BareosDb::CreateJobmediaRecord(JobControlRecord* jcr, JobMediaDbRecord* jm)
 {
-  int count;
-  char ed1[50], ed2[50], ed3[50];
-
   DbLocker _{this};
 
   /* clang-format off */
   Mmsg(cmd,
-       "UPDATE JobMedia SET LastIndex=%lu, EndFile=%lu, EndBlock=%lu, JobBytes=%s "
-       "WHERE JobId=%s AND MediaId=%s",
+       "UPDATE JobMedia SET "
+       "FirstIndex = (CASE when firstindex=0 then %lu else firstindex end), "
+       "StartBlock = (CASE when startblock=0 then %lu else startblock end), "
+       "StartFile  = (CASE when startfile=0 then %lu else startfile end), "
+       "LastIndex=%lu, EndFile=%lu, EndBlock=%lu, JobBytes=%llu "
+       "WHERE JobId=%lu AND MediaId=%lu",
+       jm->FirstIndex,
+       jm->StartBlock,
+       jm->StartFile,
        jm->LastIndex,
        jm->EndFile,
        jm->EndBlock,
-       edit_uint64(jm->JobBytes, ed3),
-       edit_int64(jm->JobId, ed1),
-       edit_int64(jm->MediaId, ed2));
```

Comment:
Why did we have these edit_xxx calls here?

Reply:
direct heritage from the first ever commit, no specific reason mentioned.

All variables around it use the dedicated placeholders, there is no obvious reason for not changing those too.

my guess would be that it used to be useful back when 32-bit systems were supported.